apps/sms/services/js/bridge_service_mixin.js

```diff
@@ -0,0 +1,81 @@
+/*global bridge */
+/*jshint esnext: true */
+
+'use strict';
+
+(function(exports) {
+
+  const priv = Object.freeze({
+    methods: Symbol('methods'),
+    streams: Symbol('streams'),
+    events: Symbol('events'),
+    service: Symbol('service')
+  });
+
+  /**
+   * This mixin allows to expose easily a service from an object.
+   *
+   * This needs the following properties to work properly:
+   *
+   * - SERVICE_NAME holds the service name.
+   * - METHODS holds the methods that will be exposed by the service.
+   * - STREAMS holds the methods that handle streams exposed by the service.
+   *
+   * EVENTS, METHODS and STREAMS are all optional.
+   */
+  var serviceMixin = {
+    initService() {
+      if (!('bridge' in self)) {
+        importScripts('/lib/bridge.js');
+      }
+
+      var service = bridge.service(this[priv.name]);
+
+      this[priv.methods].forEach((exposedMethod) => {
+        service.method(exposedMethod, this[exposedMethod].bind(this));
+      });
+
+      this[priv.streams].forEach((exposedStream) => {
+        service.stream(exposedStream, this[exposedStream].bind(this));
+      });
+
+      this[priv.service] = service;
+      return service;
```

Comment:
mmm, I'd rather not return service from here, I think it's better to keep it private, otherwise we can broadcast not allowed events, define not supported methods and etc. What do you think?